_pytest/skipping.py

```diff
@@ -60,22 +60,31 @@ def nop(*args, **kwargs):
                             )
 
 
-class MarkEvaluator:
+class MarkEvaluator(object):
     def __init__(self, item, name):
         self.item = item
-        self.name = name
-
-    @property
-    def holder(self):
-        return self.item.keywords.get(self.name)
+        self._marks = None
+        self._mark = None
+        self._repr_name = name
```

Comment:
Perhaps this should be `self._mark_name`? Seems clearer to me. "repr" is too close to Python's own `__repr__` which might be confusing.